DrawingEngineFactory/DrawingEngineFactory.cpp

```diff
@@ -0,0 +1,20 @@
+#include "DrawingEngineFactory.h"
+
+DrawingEngineFactory::DrawingEngineFactory(any engineI) : ClassFactory() {
+	 setVar(engine, engineI, "engine");
+}
+
+shared_ptr<DrawingEngine> DrawingEngineFactory::createEngine(int widthI, int heightI, char wallI, char routeI) {
```

Comment:
this could and should be a static method. This way you don't have to create the object of the factory class before calling createEngine.

Also you should use enum class as the "engine" type so that it is much easier to understand which drawing englne will be created.